packages/vuetify/src/components/VTextField/VTextField.ts

```diff
@@ -55,6 +55,7 @@ export default baseMixins.extend<options>().extend({
       default: '$vuetify.icons.clear',
     },
     counter: [Boolean, Number, String],
```

Comment:
Simply add `Function` to the list instead of creating a new prop?
`counter: [Boolean, Number, String, Function],`

Reply:
Attention, the `counter` property in `VTextField` is the value of the `maxLength` not the value itself!
I want to name the `counterLength` `counterValue`, but there is already the `counterValue` method and I didn't want to conflict with it.
If someone in the vuetify core team would accept to use `counterValue` instead of `counterLength`, I can rewrite the PR if this is wanted

Reply:
Would say name it something like `counterText`.

Reply:
`counterText` suggests a bit more than just a length:

`${counterValue} / 30` vs `${counterText}`

Maybe slot would be a good idea? You could add literally anything, including colors, icons etc. More powerful but also more to typing

```html
<v-text-field #counter="{ value, max }">{{ someFun(value) }} / {{ max }}</v-text-field>
```

```html
<v-text-field>
  <template #counter="{ value, max }">
    <v-icon class="red">foo</v-icon> 
    {{ value < 10 : 'too short' : 'too long'}}, max: {{ max }}</span>
  </template>
</v-text-field>
```

Async would need an additional variable in data, but i don't think it's a frequent case.

Reply:
ah, I was under the impression it was replacing the whole thing not just the starting value, in which what would be the point of making it a function (you could just bind a string or number). 
Maybe a slot would be more valuable, providing the most flexibility.

Reply:
I **try** not to make any breaking changes

Reply:
It wouldn't be a breaking change